vcf2fhir/fhir_helper.py

```diff
@@ -597,3 +547,25 @@ def generate_final_json(self):
     def export_fhir_json(self, output_filename):
         with open(output_filename, 'w') as fp:
             json.dump(self.fhir_json, fp, indent=4)
+
+
+def _append(value_to, value_from, list_of_sub_keys, code):

```

Comment:
Instead of calling it append we can give it more meaningful name. Actually, I would prefer to have a method that creates a required dictionary from another object and returns that dictionary. Actual append can happen in the caller (`generate_final_json`) to make it more readable. We should avoid creating two different kind of dictionary in same method (eg. category and coding here), even though number of code lines are reduced, readability is compromised. Function should target only one task. If you follow what I said, function signature will be much simplified and other can just understand purpose of function just from its signature.

Reply:
so something like this 
```
def get_block(value_from, list_of_sub_keys, code):
    value_to = OrderedDict()
    key = None
    is_category = False
    if code is 0:
        key = 'category'
        is_category = True
    elif code is 1:
        key = 'code'
    elif code is 2:
        key = 'valueCodeableConcept'
    for k in list_of_sub_keys:
        if is_category:
            value_to[k] = value_from[key][0]['coding'][0][k]
        elif k in value_from[key]['coding'][0].keys():
            value_to[k] = value_from[key]['coding'][0][k]
    return value_to # here you returning that block/dic
```

Reply:
```
def get_block(value_from,, key (category/code/valueCodeableConcept)):
   value_to = OrderedDict()

   value_to[system] = value_from[key]['coding'][0][k]
   value_to[code] = value_from[key]['coding'][0][k]
   ....
   return value_to # here you returning that block/dic
```

We can add another similar method for category (no if else). if something requires only two keys `system`, `code` means that is different type of block so we can create method which only adds `system` and `code`. Aim is to have simple to understand private method.